app-web/plugins/gatsby-source-github-all/utils/fetchSourceGithub.js

```diff
@@ -224,14 +224,33 @@ const applyBaseMetadata = (
   };
 };
 
+/**
+ * filters files by the context directories from source
+ * @param {Array} files the files
+ * @param {Object} contextDir array of paths to get files in a repo
+ */
+const filterFilesByContext = (files, contextDir) => {
+  const targetPaths = contextDir.map(dir => {
+    return dir.charAt(0) === '/' ? dir.substring(1) : dir;
+  });
+  // now only return files in the target context dir array
+  const contextFiles = files.filter(file => {
+    return targetPaths.some(path => file.path.indexOf(path) >= 0);
```

Comment:
this will cause any paths to validate to true
if path is `docs` and file.path is `/foo/bar/docs` this will pass this test. 
We should explictily define context directories based on the root. if the example path wants to be set as a context dir
it should have a path in the registry set as `/foo/bar/docs`

Reply:
i'd suggest either one of two options
`indexOf(path) === 0` or create a regex object from file.path and test it 
```javascript
const re = new RegExp(`^${file.path}`);
return re.test(path);
```

Reply:
as the file paths are all relative to root, will get rid if extra `/` from contextDir and match with `indexOf(path) === 0`